### ISSUE TITLE
module 'connman_dbus._service' not found

### ISSUE THREAD

**Issue by mscharley** [stefano-m/lua-connman_dbus#1]
I'm trying to get `stefano-m/awesome-connman_widget` working on my computer but after installing it using `luarocks install --local connman_widget` I get the following error:

![image](https://user-images.githubusercontent.com/336509/68639145-c09c8c00-0557-11ea-82de-3e15455acc2b.png)

Looking in the luarocks folder all I can see inside the `connman_dbus` folder is `init.lua`, not any of the other files in the src folder of this repo. Did I get a bad installation somehow?

**Comment by mscharley** [comment]
Install log:

```
Installing https://luarocks.org/connman_widget-0.3.0-2.src.rock               
Missing dependencies for connman_widget 0.3.0-2:                   
   connman_dbus >= 0.3.0, < 0.6 (not installed)                               
                                                                               
connman_widget 0.3.0-2 depends on connman_dbus >= 0.3.0, < 0.6 (not installed)
Installing https://luarocks.org/connman_dbus-0.5.0-1.src.rock                  
                                                                               
connman_dbus 0.5.0-1 is now installed in /home/momo/.luarocks (license: Apache v2.0)
                                                                               
connman_widget 0.3.0-2 is now installed in /home/momo/.luarocks (license: GPL v3)
```

**Comment by oliverjkb** [comment]
Same Problem here. 

Luarocks List:
```
~# luarocks list

Installed rocks:
----------------

connman_dbus
   0.5.0-1 (installed) - /usr/local/lib/luarocks/rocks

connman_widget
   0.3.0-2 (installed) - /usr/local/lib/luarocks/rocks

dbus_proxy
   0.10.0-1 (installed) - /usr/local/lib/luarocks/rocks

lgi
   0.9.2-1 (installed) - /usr/local/lib/luarocks/rocks
```
Luarocks info:
```
~# luarocks --version
/usr/bin/luarocks 2.4.2
LuaRocks main command-line interface
```

Lua Version:
```
~# lua -v
Lua 5.3.3  Copyright (C) 1994-2016 Lua.org, PUC-Rio
``` 
Running on ubuntu 19.10.

I'm fairly fresh to Lua, so I cannot tell the reason this is happening. 

However, saving `_service.lua` and `_technology.lua` in the directory  in your error message ( i.e. just `wget` the raw files from this repo) should fix the issue, although probably not in a recommended fashion ^^ 

**EDIT**
I just downloaded both, Version 0.5.0-1 and 0.4.0-2 from luarocks.org and unpacked them. 

```
ls -a connman_dbus-0.*.0-*/lua-connman_dbus/src/connman_dbus                                                                                                0s 

connman_dbus-0.4.0-2/lua-connman_dbus/src/connman_dbus:
init.lua

connman_dbus-0.5.0-1/lua-connman_dbus/src/connman_dbus:
_service.lua
_technology.lua
init.lua
```
So the file _are_ present in the package, it seems.

**Comment by stefano-m** [comment]
Hello, and thanks for reporting this.

I am currently not using luarocks so it may take some time for me to look at this issue.

Can you confirm whether all versions are affected or just one/some please? It may be an error in creating the rockspec files.

Thanks

**Comment by mscharley** [comment]
If I run `luarocks install --local connman_dbus 0.4.0-2` then it replaces the 0.5.0 release and everything works.

Having a look, I'd guess this is the issue:

https://github.com/stefano-m/lua-connman_dbus/blob/master/rockspec/connman_dbus-0.5.0-1.rockspec#L22-L24

In 0.5.0 you've split the main init.lua into three files, but they aren't listed in the rockspec.

**Comment by stefano-m** [comment]
You are correct @mscharley 

I think this is now fixed and I published https://luarocks.org/modules/stefano-m/connman_dbus/0.5.0-2

If you would be so kind to confirm that now it works, I will close this issue.

Thanks for your patience!

**Comment by mscharley** [comment]
I installed 0.5.0-2 and it's working for me now. Thanks!

**Comment by stefano-m** [comment]
Closed by b42c477
